plugins/PrivacyManager/Tracker/RequestProcessor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\PrivacyManager\Tracker;
+
+use Piwik\Common;
+use Piwik\SettingsPiwik;
+use Piwik\Tracker\Request;
+use Piwik\Tracker;
+use Piwik\Plugins\PrivacyManager\Config as PrivacyManagerConfig;
+
+class RequestProcessor extends Tracker\RequestProcessor
+{
+    public function manipulateRequest(Request $request)
+    {
+        $privacyConfig = new PrivacyManagerConfig();
+
+        if ($privacyConfig->anonymizeUserId) {
+            $userId = $request->getParam('uid');
+            if ($this->isValueSet($userId)) {
+                $userIdAnonymized = self::anonymizeUserId($userId);
+                $request->setParam('uid', $userIdAnonymized);
+            }
+        }
+
+        if ($privacyConfig->anonymizeOrderId) {
+            $orderId = $request->getParam('ec_id');
+            if ($this->isValueSet($orderId)) {
+                $orderIdAnonymized = sha1(Common::getRandomInt() . $orderId . time() . SettingsPiwik::getSalt());
+                $request->setParam('ec_id', $orderIdAnonymized);
+            }
+        }
+    }
+
+    public static function anonymizeUserId($userId)
+    {
+        // pseudo anonymization as we need to make sure to always generate the same UserId for the same original UserID
+        return sha1($userId . SettingsPiwik::getSalt());
+    }
+
+    private function isValueSet($value)
+    {
+        return $value !== '' && $value !== false && $value !== null;
+    }
```

Comment:
Seeing this code in a lot of places now, think it should be moved to `Piwik\Common` or something like that?